Content.Shared/Atmos/EntitySystems/SharedGasTileOverlaySystem.cs

```diff
@@ -63,10 +63,16 @@ public bool Equals(GasOverlayData other)
                 if (FireState != other.FireState)
                     return false;
 
-                for (var i = 0; i < Opacity.Length; i++)
+                if (Opacity?.Length != other.Opacity?.Length)
+                    return false;
+
+                if (Opacity != null && other.Opacity != null)
                 {
```

Comment:
This function should probably return false if one is null and the other is not null

Reply:
ah never mind, the length check does that already

Reply:
Though AFAIK i the array isn't null then its length should always be `VisibleGasId.Length`, so it could just be a null-check anyways

Reply:
I left the redundant check in there since the struct is public, so it could get initialized with something else.
If a constant was used internally that would make sense though.